docs/installation/Installing_ACME_Responder.md

```diff
@@ -0,0 +1,186 @@
+```
+
+To enable the profile, execute the following command:
+
+```
+$ pki -u caadmin -w Secret.123 ca-profile-enable acmeServerCert
+```
+
+## Installing ACME Responder
+
+To install the ACME responder on PKI server, execute the following command:
+
+```
+$ pki-server acme-create
+```
+
+The command will create the initial configuration files in /etc/pki/pki-tomcat/acme folder.
+
+## Configuring ACME Responder Database
+
+The database configuration for the ACME responder is located at /etc/pki/pki-tomcat/acme/database.json.
+
+To use an in-memory database, copy the sample configuration file with the following command:
+
+```
+$ cp /usr/share/pki/acme/conf/database/in-memory/database.json \
+    /etc/pki/pki-tomcat/acme/database.json
+```
+
+Alternatively, edit the file as follows:
+
+```
+{
+    "class": "org.dogtagpki.acme.database.InMemoryDatabase"
+}
+```
+
+Currently there are no parameters to configure.
+
+See also [Configuring ACME Responder](https://www.dogtagpki.org/wiki/Configuring_ACME_Responder).
+
+## Configuring ACME Responder Backend
+
+The backend configuration for the ACME responder is located at /etc/pki/pki-tomcat/acme/backend.json.
+
+To use the CA subsystem as the backend for the ACME responder,
+copy the sample configuration with the following command:
+
+```
+$ cp /usr/share/pki/acme/conf/backend/pki/backend.json \
+    /etc/pki/pki-tomcat/acme/backend.json
+```
+
+Alternatively, edit the file as follows:
+
+```
+{
+    "class": "org.dogtagpki.acme.backend.PKIBackend",
+    "parameters": {
+        "url": "https://localhost:8443",
+        "profile": "acmeServerCert",
+        "username": "caadmin",
+        "password": "Secret.123"
+    }
+}
+```
+
+Configure the parameters as needed.
+
+See also [Configuring ACME Responder](https://www.dogtagpki.org/wiki/Configuring_ACME_Responder).
+
+## Deploying ACME Responder
+
+Once everything is ready, deploy the ACME responder with the following command:
+
+```
+$ pki-server acme-deploy
+```
+
+The command will create a deployment descriptor at /etc/pki/pki-tomcat/Catalina/localhost/acme.xml.
+
+The server will start the ACME responder automatically in a few seconds.
+It is not necessary to restart PKI server.
+
+To verify that the ACME responder is running, execute the following command:
+
+```
+$ curl -s -k https://$HOSTNAME:8443/acme/directory | python -m json.tool
+{
+    "meta": {
+        "caaIdentities": [
+            "dogtagpki.org"
+        ],
+        "externalAccountRequired": false,
+        "termsOfService": "https://www.dogtagpki.org/wiki/PKI_ACME_Service",
+        "website": "https://www.dogtagpki.org"
+    },
+    "newAccount": "https://<hostname>:8443/acme/new-account",
+    "newNonce": "https://<hostname>:8443/acme/new-nonce",
+    "newOrder": "https://<hostname>:8443/acme/new-order"
+}
+```
+
+## Using ACME Responder with certbot
+
+Note that certbot does not accept self-signed CA certificate,
+so the sample operations below are executed with plain HTTP URLs.
```

Comment:
Should probably add a comment here about in production, the CA's certificate should be added to the system trust store... :)

Reply:
Hmm.. I don't think we should encourage people to try it in production right now. Maybe later when it's production-ready. We will probably need to point to another document that describes how to add into the system trust store. It's probably outside the scope of this document.

Reply:
Eh that's fair. Perhaps a big:

----

### **WARNING**

**Using ACME over http is insecure and shouldn't be done in production.**

---- 

instead? :)

Reply:
That's quite big :)
How about a box like this?
https://stackoverflow.com/questions/50544499/how-to-make-a-styled-markdown-admonition-box-in-a-github-gist

Reply:
Sure, anything works :-) I was using `<hr>` to separate the warning box from the rest of my comment.

Reply:
Welll, ACME over non-TLS is verboten, per RFC 8555.  So in this installation guide, let's just add a brief admonition that the CA cert must be in system trust store, and show all commands using HTTPS.

Reply:
I added a warning box.

Since this is a tech-preview people, will likely use self-signed CA. I don't think we should recommend adding a self-signed CA cert into the system trust store because it clutters the whole system and people may forget to remove it. It's probably better to specify a parameter (e.g. server URL) that would be visible and can be changed easily when the proper CA is ready.

If people wants to add a self-signed CA cert into system trust store, they can surely do that, but I don't think we should recommend it. It also requires a root privilege, which is a requirement that we want to eventually remove.